flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/SortTestPrograms.java

```diff
@@ -123,4 +125,66 @@ public class SortTestPrograms {
                                     .build())
                     .runSql("INSERT INTO sink_t SELECT * from source_t ORDER BY a DESC LIMIT 3")
                     .build();
+
+    static final TableTestProgram SORT_ASC =
+            TableTestProgram.of("sort-asc", "validates sort node by sorting integers in asc mode")
+                    .setupConfig(InternalConfigOptions.TABLE_EXEC_NON_TEMPORAL_SORT_ENABLED, true)
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema("a INT", "b VARCHAR", "c INT")
+                                    .addOption("terminating", "true")
+                                    .producedBeforeRestore(DATA)
+                                    .producedAfterRestore(Row.of(11, "a", 5), Row.of(8, "c", 9))
+                                    .build())
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink_t")
+                                    .addSchema("a INT", "b VARCHAR", "c BIGINT")
+                                    .consumedBeforeRestore(
+                                            "+I[1, a, 5]",
+                                            "+I[2, a, 6]",
+                                            "+I[3, b, 7]",
+                                            "+I[4, b, 8]",
+                                            "+I[5, c, 9]",
+                                            "+I[6, c, 10]")
+                                    .consumedAfterRestore(
+                                            "+I[1, a, 5]",
+                                            "+I[2, a, 6]",
+                                            "+I[3, b, 7]",
+                                            "+I[4, b, 8]",
+                                            "+I[5, c, 9]",
+                                            "+I[6, c, 10]")
+                                    .build())
+                    .runSql("INSERT INTO sink_t SELECT * from source_t ORDER BY a")
+                    .build();
+
+    static final TableTestProgram SORT_DESC =
+            TableTestProgram.of("sort-desc", "validates sort node by sorting integers in desc mode")
+                    .setupConfig(InternalConfigOptions.TABLE_EXEC_NON_TEMPORAL_SORT_ENABLED, true)
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema("a INT", "b VARCHAR", "c INT")
+                                    .addOption("terminating", "true")
```

Comment:
same as above

Reply:
Removed by converting to a non restore test